config/config-server/src/com/thoughtworks/go/service/ConfigRepository.java

```diff
@@ -404,8 +404,31 @@ public void run() throws Exception {
     public long getLooseObjectCount() throws Exception {
         return doLocked(new ThrowingFn<Long, GitAPIException>() {
             public Long call() throws GitAPIException {
-                return (Long) git.gc().getStatistics().get("numberOfLooseObjects");
+                return (Long) getStatistics().get("numberOfLooseObjects");
             }
         });
     }
+
+    public Properties getStatistics() throws GitAPIException {
+        // not inside a doLocked/synchronized block because we don't want to block the server status service.
+        return git.gc().getStatistics();
+    }
+
+    public Long commitCountOnMaster() throws GitAPIException, IncorrectObjectTypeException, MissingObjectException {
+        // not inside a doLocked/synchronized block because we don't want to block the server status service.
+        // we do a `git branch` because we switch branches as part of normal git operations,
+        // and we don't care about number of commits on those branches.
+        List<Ref> branches = git.branchList().call();
+        for (Ref branch : branches) {
```

Comment:
I could not find one with a quick google search and reading through the API. I'm sure there's some low level API, but this felt good enough to me.